.template-helpers/variant.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+set -e
+
+repo="$1"
+if [ -z "$repo" ]; then
+	echo >&2 "usage: $0 repo"
+	echo >&2 "   ie: $0 hylang"
+	exit 1
+fi
+
+dir="$(dirname "$(readlink -f "$BASH_SOURCE")")"
+
+IFS=$'\n'
+lines=( $(curl -sSL 'https://raw.githubusercontent.com/docker-library/official-images/master/library/'"$repo" | grep -vE '^$|^#') )
+unset IFS
+
+tags=()
+# this loop finds all the "variants", making assumptions about what they can contain
+for line in "${lines[@]}"; do
+	tag="$(echo "$line" | awk -F ': +' '{ print $1 }' | sed -r 's/[0-9.-]+//g')"
+	if [ "$tag" ]; then
+		tags+=($tag)
+	fi
+done
+
+IFS=$'\n'
+# tags can't have spaces so no "$()" is ok
+tags=($(echo "${tags[*]}" | sort -u))
+unset IFS
+
+text=
+for tag in "${tags[@]}"; do
+	if [ -f "$dir/variant-${tag}.md" ]; then
+		text+=$'\n' # give a little space
+		# because parameter expansion eats the trailing newline
+		text+="$(<"$dir/variant-${tag}.md")"$'\n'
+	fi
+done
+if [ "$text" ]; then
+	# because parameter expansion eats the trailing newline
+	text=$'\n\n'"$(<"$dir/variant.md")"$'\n'"$text"
```

Comment:
At this point, why not just skip the `echo` outside the `if` and do something like:
```bash
if [ "$text" ]; then
    echo
    cat "$dir/variant.md"
    echo -n "$text"
fi
```